### PR TITLE
Add Android support 2

### PR DESCRIPTION
**Pull request by dylanaraps**
This PR is the same as the last one, it just has all of the new changes from master (350~ Commits). A lot of the bugs from the previous PR should be fixed now and I'd love it if you guys could help me test it out. I've only got one Android device in my house and I'd like to make sure it works for as many people as possible.

Known bugs:

- [x] Ascii art padding is broken with ascii art that uses the unicode block char. (Android, Manjaro).
    - Cause: Android devices don't seem to have any locale related envars set(?) so neofetch is using `LANG=C` which incorrectly calculates the padding when unicode block chars are involved.
    - Solution: Assume block char is `3` chars wide when calculating padding.
- [ ] `./neofetch` doesn't work.
    - Workaround: `bash ./neofetch`
    - `env` doesn't exist on android.
- [x] Fixed missing CPU name. @konimex 
- [x] Broken `uptime` output.
- [x] Added model output. @konimex 

Old PR: #240

### CODE REVIEW

**Comment by dylanaraps** [comment]
To fix the CPU speed issue in the previous PR I need to know the output of this command:

```sh
ls /sys/devices/system/cpu/cpu0/cpufreq/
```

**Comment by konimex** [comment]
You should consider adding Busybox as a dependency for Android, as (interesting in my case because `awk` doesn't exist in default Android)
This is the output without Busybox (and the files in `cpufreq`).
nobox

This is the output after Busybox (with `--test`).
android1

**Comment by dylanaraps** [comment]
Will do, thanks for testing.

**Comment by dylanaraps** [comment]
Can you send me the output of `cat /proc/cpuinfo` on your device?

**Comment by konimex** [comment]
This is the result:
`processor       : 0
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 1
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 2
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 3
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 4
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 5
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 6
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1

processor       : 7
Features        : fp asimd evtstrm aes pmull sha1 sha2 crc32
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 1`

**Comment by dylanaraps** [comment]
It looks like we'll have to find another way to get the CPU name on Andoid since it isn't in `/proc/cpuinfo`

**Comment by dylanaraps** [comment]
Output from my friend's phone:

14194384_1056800781064189_834584356_n

**Comment by konimex** [comment]
Actually, to fix some of the problems (no name output on CPU), we can use `/system/build.prop`. Some of the important Android system information (phone model, CPU) are stored there.

**Comment by dylanaraps** [comment]
I recently discovered `termux` which is an open source terminal emulator/linux environment for android.

Website: https://termux.com/
Source code: https://github.com/termux

#322 should work relatively bug free on Termux. My current thoughts are to officially support a single  environment on Android (Termux) and either list Termux as a dependency or recommend it in the readme. 

There are too many environments on android and due to the nature of neofetch (looking for files, commands etc) it'll be impossible to support them all.  

What are your thoughts on this?

**Comment by konimex** [comment]
Actually, I don't think Termux is needed at all (well, for non-rooted devices, Termux is a comfortable solution since it has built-in Busybox environment so no hassle). Android has begun trying to get its shit together since 6.0 but Toybox hasn't caught up with Busybox yet, so Busybox is needed.

But since Termux's built in `ps` is much saner than Android's, I think just a dependency on Busybox will be fine, and add that it is recommended to install Termux since it has Busybox and a good `ps` built-in.

**Comment by dylanaraps** [comment]
Sounds good.

I'll fix the merge conflicts and reopen this PR with the latest changes from master. Then we can test the current status of this PR and fix any other bugs that come up.

**Comment by dylanaraps** [comment]
Fixed all merge conflicts and rebased with master. 

Please test the branch now.

**Comment by konimex** [comment]
screenshot_20161019-212538

Looks good enough.

**Comment by dylanaraps** [comment]
I'm assuming that `./neofetch` still doesn't work?

Can you also try with `neofetch --config off --test`?

**Comment by konimex** [comment]
Yep, `./neofetch` doesn't work. Could this be related with the shebang on the first line since Android doesn't have `/usr/bin/env` ?

This is the second command (+ `--disable publicip`)

hello

**Comment by dylanaraps** [comment]
This looks bug free!

I'll test some more at home and probably merge this later today.

> Could this be related with the shebang on the first line since Android doesn't have /usr/bin/env ?

Yup

Thanks!

File: neofetch [hunk 1]
```diff
@@ -452,6 +465,8 @@ getpackages() {
             packages="$(pkg list | wc -l)"
         ;;
     esac
+
+    [ "$packages" == 0 ] && unset packages
```
Comment (dylanaraps) [inline neofetch]: We may not need this.

File: neofetch [hunk 2]
```diff
@@ -1717,8 +1736,8 @@ getbattery() {
                     *) index="$battery_num" ;;
                 esac
 
-                batteries=("$(cat /sys/class/power_supply/BAT$battery_num/capacity)")
-                battery_state=("$(cat /sys/class/power_supply/BAT${battery_num}/status)")
+                batteries=($(cat /sys/class/power_supply/battery/capacity /sys/class/power_supply/BAT${battery_num}/capacity))
```
Comment (dylanaraps) [inline neofetch]: This would be better as a `distro` case statement.

File: neofetch [hunk 3]
```diff
@@ -2063,14 +2082,18 @@ getascii() {
...
     ascii_strip="${ascii_strip//'\'}"
 
-    # Get lines/columns of the ascii file in pure bash.
+    # Get length of the ascii file.
     ascii_length=0
-    lines=1
-    while IFS='\n' read -r line 2>/dev/null; do
-        [ ${#line} -gt "$ascii_length" ] && ascii_length=${#line}
-        lines=$((lines+=1))
+    while IFS='' read -r line 2>/dev/null; do
```
Comment (dylanaraps) [inline neofetch]: This ovewrites some bug fixes

**Comment by dylanaraps** [comment]
I pushed some fixes, can you test again with `neofetch --config off --test`?

**Comment by konimex** [comment]
Screen2

**Comment by dylanaraps** [comment]
👍

**Comment by dylanaraps** [comment]
The block chars in the ascii logo break the output in `Material Terminal` and `Terminal Emulator for Android`. Other logos render just fine, any ideas?



333p333

**Comment by konimex** [comment]
The block characters probably isn't ASCII so the render broke.

**Comment by dylanaraps** [comment]
Yeah, the only solution is to create new ascii art that uses regular characters. I've attempted it and none of my attempts look as nice as this blocky art.

**Comment by dylanaraps** [comment]
Merging this now.

**Pull request merged**